pkg/productinfo/alibaba/productinfo_alibaba.go

```diff
@@ -0,0 +1,244 @@
+
+	regions, err := e.GetRegions()
+	if err != nil {
+		return nil, err
+	}
+
+	for region := range regions {
+		request := ecs.CreateDescribeInstanceTypesRequest()
+		request.RegionId = region
+
+		vmSizes, err := e.client.DescribeInstanceTypes(request)
+		if err != nil {
+			return nil, err
+		}
+		instanceTypes := vmSizes.InstanceTypes.InstanceType
+		for _, v := range instanceTypes {
+			switch attribute {
+			case productinfo.Cpu:
+				valueSet[productinfo.AttrValue{
+					Value:    float64(v.CpuCoreCount),
+					StrValue: fmt.Sprintf("%v", v.CpuCoreCount),
+				}] = ""
+			case productinfo.Memory:
+				valueSet[productinfo.AttrValue{
+					Value:    v.MemorySize,
+					StrValue: fmt.Sprintf("%v", v.MemorySize),
+				}] = ""
+			}
+		}
+
+		for attr := range valueSet {
+			values = append(values, attr)
+		}
+	}
+	log.Debugf("found %s values: %v", attribute, values)
+	return values, nil
+}
+
+// GetProducts retrieves the available virtual machines based on the arguments provided
+func (e *AlibabaInfoer) GetProducts(regionId string) ([]productinfo.VmInfo, error) {
+	log.Debugf("getting product info [region=%s]", regionId)
+	var vms []productinfo.VmInfo
+
+	request := ecs.CreateDescribeInstanceTypesRequest()
+	request.RegionId = regionId
+
+	vmSizes, err := e.client.DescribeInstanceTypes(request)
+	if err != nil {
+		return nil, err
+	}
+	instanceTypes := vmSizes.InstanceTypes.InstanceType
+	for _, instanceType := range instanceTypes {
+		vms = append(vms, productinfo.VmInfo{
+			Type:    instanceType.InstanceTypeId,
+			Cpus:    float64(instanceType.CpuCoreCount),
+			Mem:     instanceType.MemorySize,
+			Gpus:    float64(instanceType.GPUAmount),
+			NtwPerf: fmt.Sprintf("%.1f Gbit/s", float64(instanceType.InstanceBandwidthRx)/1024000),
+		})
+	}
+	log.Debugf("found vms: %#v", vms)
+	return vms, nil
+}
+
+// GetZones returns the availability zones in a region
+func (e *AlibabaInfoer) GetZones(region string) ([]string, error) {
+	var zones []string
+
+	request := ecs.CreateDescribeZonesRequest()
+	request.RegionId = region
+	response, err := e.client.DescribeZones(request)
+	if err != nil {
+		return nil, err
+	}
+	zonesResponse := response.Zones.Zone
+
+	for _, zone := range zonesResponse {
+		zones = append(zones, zone.ZoneId)
+	}
+	return zones, nil
+}
+
+// GetRegions returns a map with available regions
+func (e *AlibabaInfoer) GetRegions() (map[string]string, error) {
+	var RegionIdMap = make(map[string]string)
+	request := ecs.CreateDescribeRegionsRequest()
+	request.AcceptLanguage = "en-US"
+	response, err := e.client.DescribeRegions(request)
+	if err != nil {
+		return nil, err
+	}
+	regions := response.Regions.Region
+	for _, region := range regions {
+		RegionIdMap[region.RegionId] = region.LocalName
+	}
+	return RegionIdMap, nil
+}
+
+// HasShortLivedPriceInfo - Alibaba doesn't have frequently changing prices
+func (e *AlibabaInfoer) HasShortLivedPriceInfo() bool {
+	return false
```

Comment:
according to [this](https://www.alibabacloud.com/help/doc-detail/52088.htm) documentation, Alibaba spot instances and pricing is very similar to AWS. It has a constantly changing market price and bidding, so this method should return true and the (spot price) implementation should work in a way that it's queried frequently instead of once per day.